src/ngraph/op/fused/clamp.cpp

```diff
@@ -0,0 +1,55 @@
+//*****************************************************************************
+// Copyright 2017-2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#include "ngraph/op/fused/clamp.hpp"
+
+#include "ngraph/builder/make_constant.hpp"
+#include "ngraph/op/maximum.hpp"
+#include "ngraph/op/minimum.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+op::Clamp::Clamp(const shared_ptr<Node>& data, const double min, const double max)
+    : FusedOp("Clamp", {data})
+    , m_min{min}
+    , m_max{max}
+{
+    NODE_VALIDATION_CHECK(
```

Comment:
I think it's okay for now, but I feel like this check should be in `validate_and_infer_types`. It's hard to shoehorn it in there, though, because that function has to do the work of checking the decomposed graph as well, and that's handled by the base class.

Reply:
We have pre- and post-validation hooks for these kinds of checks. https://github.com/NervanaSystems/ngraph/blob/master/src/ngraph/op/util/fused_op.hpp#L40
Can we override and use those?

Reply:
Nice- didn’t realize that was there. Should be the right place for this check.